UnityProject/Assets/Scripts/Items/Tool/AdvancedLightReplacer.cs

```diff
@@ -0,0 +1,139 @@
+		{
+			if (lightTuner && interaction.IsAltClick)
+			{
+				LightTunerWindowOpen();
+				return;
+			}
+			lightTuner = !lightTuner;
+			var text = lightTuner ? "will tune lights now." : "will replace lights now.";
+			Chat.AddExamineMsg(interaction.Performer, $"this {gameObject.ExpensiveName()} {text}");
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (gameObject.PickupableOrNull()?.ItemSlot == null) return false;
+			return interaction.TargetObject != null && DefaultWillInteract.Default(interaction, side);
+		}
+
+		public bool WillInteract(HandActivate interaction, NetworkSide side)
+		{
+			return DefaultWillInteract.Default(interaction, side);
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (Validations.HasItemTrait(interaction.TargetObject, CommonTraits.Instance.LightBulb) ||
+			    Validations.HasItemTrait(interaction.TargetObject, CommonTraits.Instance.LightTube))
+			{
+				storage.ServerTryAdd(interaction.TargetObject);
+				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You watch as the tool automatically pulls out a mechanical arm that slots in the {interaction.TargetObject.ExpensiveName()}",
+					$"{interaction.PerformerPlayerScript.visibleName} slots in the {interaction.TargetObject.ExpensiveName()} using the {gameObject.ExpensiveName()}.");
+				return;
+			}
+			if (interaction.TargetObject.TryGetComponent<LightSource>(out var source) == false) return;
+			if (lightTuner)
+			{
+				SetLightColors(source);
+			}
+			else
+			{
+				var target = storage.GetTopOccupiedIndexedSlot();
+				if (target == null)
+				{
+					source.TryReplaceBulb(interaction);
+					return;
+				}
+				if (target.ItemAttributes.GetTraits().Contains(source.TraitRequired) == false) return;
+				source.TryReplaceBulb(interaction);
+				AddLightToFixture(target, source, interaction);
+				Chat.AddExamineMsg(interaction.Performer, "You replace the light-bulb with another one.");
+			}
+		}
+
+		private void AddLightToFixture(ItemSlot target, LightSource source, HandApply interaction)
+		{
+			target.ItemStorage.ServerTryRemove(target.ItemObject, false, interaction.PerformerPlayerScript.AssumedWorldPos);
+			source.TryAddBulb(target.ItemObject);
+		}
+
+		private void LightTunerWindowOpen()
+		{
+			UIManager.Instance.GlobalColorPicker.EnablePicker(SetColorToTune);
+		}
+
+		private void SetColorToTune(Color newColor)
+		{
+			currentColor = newColor;
+		}
+
+		private void SetLightColors(LightSource source)
+		{
+			source.ONColour = currentColor;
+		}
+
+		public string HoverTip()
+		{
+			StringBuilder statusText = new StringBuilder();
+			statusText.AppendLine($"Lighter Tuner On: {lightTuner}");
+			statusText.AppendLine($"Slots used: {storage.GetOccupiedSlots().Count} out of {storage.ItemStorageStructure.IndexedSlots}");
+			return statusText.ToString();
+		}
+
+		public string CustomTitle() { return null; }
```

Comment:
Use arrow syntax if you are going to return in the same line

Reply:
done